src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -304,8 +306,13 @@ public function denormalize($data, string $type, string $format = null, array $c
         $normalizedData = $this->prepareForDenormalization($data);
         $extraAttributes = [];
 
-        $reflectionClass = new \ReflectionClass($type);
-        $object = $this->instantiateObject($normalizedData, $type, $context, $reflectionClass, $allowedAttributes, $format);
+        $instantiatorResult = $this->instantiator->instantiate($type, $normalizedData, $context, $format);
+        if (null === ($object = $instantiatorResult->getObject())) {
```

Comment:
If I understand it right, result object signature is: "if object is null, it means we could not instantiate it" may be a more semantic method such as `InstiantatorResult::hasFailed(): bool` or `InstiantatorResult::isError(): bool` could help for readability here ?